test/test_files/lsqb/lsqb_queries.test

```diff
@@ -1,5 +1,5 @@
 -DATASET CSV lsqb-sf01
--BUFFER_POOL_SIZE 1073741824
+-BUFFER_POOL_SIZE 4294967296
```

Comment:
For q3 we overestimate the cardinality of the cycle query `MATCH (person1)-[:Person_knows_Person]-(person2)-[:Person_knows_Person]-(person3)-[:Person_knows_Person]-(person1)` so when we join it with the remaining part of the query we reorder the cycle query to be on the probe side. So q3 runs quite a bit slower and goes over the old buffer pool limit.

Reply:
The plan (the problematic join is the outer/leftmost one):
```
┌────────────────────────────┐
│┌──────────────────────────┐│
││       Logical Plan       ││
│└──────────────────────────┘│
└────────────────────────────┘
┌────────────────────────────┐                                                                                                                                                                                                                         
│         PROJECTION         │                                                                                                                                                                                                                         
│   ----------------------   │                                                                                                                                                                                                                         
│   ----------------------   │                                                                                                                                                                                                                         
│       Cardinality: 1       │                                                                                                                                                                                                                         
└─────────────┬──────────────┘                                                                                                                                                                                                                         
┌─────────────┴──────────────┐                                                                                                                                                                                                                         
│         AGGREGATE          │                                                                                                                                                                                                                         
│   ----------------------   │                                                                                                                                                                                                                         
│   ----------------------   │                                                                                                                                                                                                                         
│       Cardinality: 1       │                                                                                                                                                                                                                         
└─────────────┬──────────────┘                                                                                                                                                                                                                         
┌─────────────┴──────────────┐                                                                                                                                                                                                                         
│         PROJECTION         │                                                                                                                                                                                                                         
│   ----------------------   │                                                                                                                                                                                                                         
│   ----------------------   │                                                                                                                                                                                                                         
│      Cardinality: 119      │                                                                                                                                                                                                                         
└─────────────┬──────────────┘                                                                                                                                                                                                                         
┌─────────────┴──────────────┐                                                                                                                                                                                                                         
│         HASH_JOIN          │                                                                                                                                                                                                                         
│   ----------------------   │                                                                                                                                                                                                                         
│   ----------------------   │──────────────────────────────────────────────┐                                                                                                                                                                          
│      Cardinality: 119      │                                              │                                                                                                                                                                          
└─────────────┬──────────────┘                                              │                                                                                                                                                                          
┌─────────────┴──────────────┐                                ┌─────────────┴──────────────┐                                                                                                                                                           
│         HASH_JOIN          │                                │          FLATTEN           │                                                                                                                                                           
│   ----------------------   │                                │   ----------------------   │                                                                                                                                                           
│   ----------------------   │───────────────┐                │   ----------------------   │                                                                                                                                                           
│    Cardinality: 1473696    │               │                │    Cardinality: 398749     │                                                                                                                                                           
└─────────────┬──────────────┘               │                └─────────────┬──────────────┘                                                                                                                                                           
┌─────────────┴──────────────┐ ┌─────────────┴──────────────┐ ┌─────────────┴──────────────┐                                                                                                                                                           
│          FLATTEN           │ │         PROJECTION         │ │         PROJECTION         │                                                                                                                                                           
│   ----------------------   │ │   ----------------------   │ │   ----------------------   │                                                                                                                                                           
│   ----------------------   │ │   ----------------------   │ │   ----------------------   │                                                                                                                                                           
│   Cardinality: 22015143    │ │     Cardinality: 18135     │ │    Cardinality: 398749     │                                                                                                                                                           
└─────────────┬──────────────┘ └─────────────┬──────────────┘ └─────────────┬──────────────┘                                                                                                                                                           
┌─────────────┴──────────────┐ ┌─────────────┴──────────────┐ ┌─────────────┴──────────────┐                                                                                                                                                           
│          FLATTEN           │ │           EXTEND           │ │         HASH_JOIN          │                                                                                                                                                           
│   ----------------------   │ │   ----------------------   │ │   ----------------------   │                                                                                                                                                           
│   ----------------------   │ │   (person1)-[]-(person2)   │ │   ----------------------   │───────────────────────────────                               ──────────────────────────────────────────────┐                                              
│    Cardinality: 2063730    │ │   ----------------------   │ │    Cardinality: 398749     │                                                                                                            │                                              
│                            │ │     Cardinality: 18135     │ │                            │                                                                                                            │                                              
└─────────────┬──────────────┘ └─────────────┬──────────────┘ └─────────────┬──────────────┘                                                                                                            │                                              
┌─────────────┴──────────────┐ ┌─────────────┴──────────────┐ ┌─────────────┴──────────────┐                                                                                              ┌─────────────┴──────────────┐                               
│           EXTEND           │ │      SCAN_NODE_TABLE       │ │         HASH_JOIN          │                                                                                              │         PROJECTION         │                               
│   ----------------------   │ │   ----------------------   │ │   ----------------------   │                                                                                              │   ----------------------   │                               
│   (person3)-[]-(person1)   │ │    Tables: person1._ID     │ │   ----------------------   │                                                                                              │   ----------------------   │                               
│   ----------------------   │ │        Properties :        │ │     Cardinality: 26036     │──────────────────────────────────────────────┬───────────────                                │     Cardinality: 1700      │                               
│    Cardinality: 193457     │ │   ----------------------   │ │                            │                                              │                                               │                            │                               
│                            │ │     Cardinality: 1700      │ │                            │                                              │                                               │                            │                               
└─────────────┬──────────────┘ └────────────────────────────┘ └─────────────┬──────────────┘                                              │                                               └─────────────┬──────────────┘                               
┌─────────────┴──────────────┐                                ┌─────────────┴──────────────┐                                ┌─────────────┴──────────────┐                                ┌─────────────┴──────────────┐                               
│           EXTEND           │                                │          FLATTEN           │                                │         PROJECTION         │                                │         HASH_JOIN          │                               
│   ----------------------   │                                │   ----------------------   │                                │   ----------------------   │                                │   ----------------------   │                               
│   (person3)-[]-(person2)   │                                │   ----------------------   │                                │   ----------------------   │                                │   ----------------------   │───────────────┐               
│   ----------------------   │                                │     Cardinality: 1700      │                                │     Cardinality: 1700      │                                │     Cardinality: 1700      │               │               
│     Cardinality: 18135     │                                │                            │                                │                            │                                │                            │               │               
└─────────────┬──────────────┘                                └─────────────┬──────────────┘                                └─────────────┬──────────────┘                                └─────────────┬──────────────┘               │               
┌─────────────┴──────────────┐                                ┌─────────────┴──────────────┐                                ┌─────────────┴──────────────┐                                ┌─────────────┴──────────────┐ ┌─────────────┴──────────────┐
│      SCAN_NODE_TABLE       │                                │         HASH_JOIN          │                                │         HASH_JOIN          │                                │          FLATTEN           │ │           EXTEND           │
│   ----------------------   │                                │   ----------------------   │                                │   ----------------------   │                                │   ----------------------   │ │   ----------------------   │
│    Tables: person3._ID     │                                │   ----------------------   │                                │   ----------------------   │                                │   ----------------------   │ │   (city3)-[]->(country)    │
│        Properties :        │                                │     Cardinality: 1700      │───────────────┐                │     Cardinality: 1700      │───────────────┐                │     Cardinality: 1700      │ │   ----------------------   │
│   ----------------------   │                                │                            │               │                │                            │               │                │                            │ │     Cardinality: 1343      │
│     Cardinality: 1700      │                                │                            │               │                │                            │               │                │                            │ │                            │
└────────────────────────────┘                                └─────────────┬──────────────┘               │                └─────────────┬──────────────┘               │                └─────────────┬──────────────┘ └─────────────┬──────────────┘
                                                              ┌─────────────┴──────────────┐ ┌─────────────┴──────────────┐ ┌─────────────┴──────────────┐ ┌─────────────┴──────────────┐ ┌─────────────┴──────────────┐ ┌─────────────┴──────────────┐
                                                              │          FLATTEN           │ │           EXTEND           │ │          FLATTEN           │ │           EXTEND           │ │           EXTEND           │ │      SCAN_NODE_TABLE       │
                                                              │   ----------------------   │ │   ----------------------   │ │   ----------------------   │ │   ----------------------   │ │   ----------------------   │ │   ----------------------   │
                                                              │   ----------------------   │ │   (city1)-[]->(country)    │ │   ----------------------   │ │   (city2)-[]->(country)    │ │   (person3)-[]->(city3)    │ │     Tables: city3._ID      │
                                                              │     Cardinality: 1700      │ │   ----------------------   │ │     Cardinality: 1700      │ │   ----------------------   │ │   ----------------------   │ │        Properties :        │
                                                              │                            │ │     Cardinality: 1343      │ │                            │ │     Cardinality: 1343      │ │     Cardinality: 1700      │ │   ----------------------   │
                                                              │                            │ │                            │ │                            │ │                            │ │                            │ │     Cardinality: 1343      │
                                                              └─────────────┬──────────────┘ └─────────────┬──────────────┘ └─────────────┬──────────────┘ └─────────────┬──────────────┘ └─────────────┬──────────────┘ └────────────────────────────┘
                                                              ┌─────────────┴──────────────┐ ┌─────────────┴──────────────┐ ┌─────────────┴──────────────┐ ┌─────────────┴──────────────┐ ┌─────────────┴──────────────┐                               
                                                              │           EXTEND           │ │      SCAN_NODE_TABLE       │ │           EXTEND           │ │      SCAN_NODE_TABLE       │ │      SCAN_NODE_TABLE       │                               
                                                              │   ----------------------   │ │   ----------------------   │ │   ----------------------   │ │   ----------------------   │ │   ----------------------   │                               
                                                              │   (person1)-[]->(city1)    │ │     Tables: city1._ID      │ │   (person2)-[]->(city2)    │ │     Tables: city2._ID      │ │    Tables: person3._ID     │                               
                                                              │   ----------------------   │ │        Properties :        │ │   ----------------------   │ │        Properties :        │ │        Properties :        │                               
                                                              │     Cardinality: 1700      │ │   ----------------------   │ │     Cardinality: 1700      │ │   ----------------------   │ │   ----------------------   │                               
                                                              │                            │ │     Cardinality: 1343      │ │                            │ │     Cardinality: 1343      │ │     Cardinality: 1700      │                               
                                                              └─────────────┬──────────────┘ └────────────────────────────┘ └─────────────┬──────────────┘ └────────────────────────────┘ └────────────────────────────┘                               
                                                              ┌─────────────┴──────────────┐                                ┌─────────────┴──────────────┐                                                                                             
                                                              │      SCAN_NODE_TABLE       │                                │      SCAN_NODE_TABLE       │                                                                                             
                                                              │   ----------------------   │                                │   ----------------------   │                                                                                             
                                                              │    Tables: person1._ID     │                                │    Tables: person2._ID     │                                                                                             
                                                              │        Properties :        │                                │        Properties :        │                                                                                             
                                                              │   ----------------------   │                                │   ----------------------   │                                                                                             
                                                              │     Cardinality: 1700      │                                │     Cardinality: 1700      │                                                                                             
                                                              └────────────────────────────┘                                └────────────────────────────┘                                                    
```